core_perception/ekf_localizer/include/ekf_localizer/ekf_localizer.hpp

```diff
@@ -0,0 +1,238 @@
+#include <geometry_msgs/msg/pose_with_covariance_stamped.hpp>
+#include <geometry_msgs/msg/twist_stamped.hpp>
+#include <geometry_msgs/msg/twist_with_covariance_stamped.hpp>
+#include <geometry_msgs/msg/transform_stamped.hpp>
+#include <std_msgs/msg/float64.hpp>
+#include <std_msgs/msg/float64_multi_array.hpp>
+#include <tf2/utils.h>
+#include <tf2/LinearMath/Quaternion.h>
+#include <tf2_ros/transform_listener.h>
+#include <tf2_ros/transform_broadcaster.h>
+
+#include "amathutils_lib_ros2/kalman_filter.hpp"
+#include "amathutils_lib_ros2/time_delay_kalman_filter.hpp"
+#include <carma_ros2_utils/carma_lifecycle_node.hpp>
+
+namespace ekf_localizer{
+    
+class EKFLocalizer: public carma_ros2_utils::CarmaLifecycleNode
+{
+    private:
+
+    rclcpp::Publisher<geometry_msgs::msg::PoseStamped>::SharedPtr pub_pose_;                           //!< @brief ekf estimated pose publisher
+    rclcpp::Publisher<geometry_msgs::msg::PoseWithCovarianceStamped>::SharedPtr pub_pose_cov_;         //!< @brief estimated ekf pose with covariance publisher
+    rclcpp::Publisher<geometry_msgs::msg::TwistStamped>::SharedPtr pub_twist_;                         //!< @brief ekf estimated twist publisher
+    rclcpp::Publisher<geometry_msgs::msg::TwistWithCovarianceStamped>::SharedPtr pub_twist_cov_;       //!< @brief ekf estimated twist with covariance publisher
+    rclcpp::Publisher<std_msgs::msg::Float64MultiArray>::SharedPtr pub_debug_;                         //!< @brief debug info publisher
+    rclcpp::Publisher<geometry_msgs::msg::PoseStamped>::SharedPtr pub_measured_pose_;                  //!< @brief debug measurement pose publisher
+    rclcpp::Publisher<std_msgs::msg::Float64>::SharedPtr pub_yaw_bias_;                                //!< @brief ekf estimated yaw bias publisher
+    
+    rclcpp::Subscription<geometry_msgs::msg::PoseWithCovarianceStamped>::SharedPtr sub_initialpose_;     //!< @brief initial pose subscriber
+    rclcpp::Subscription<geometry_msgs::msg::PoseWithCovarianceStamped>::SharedPtr sub_pose_with_cov_;   //!< @brief measurement pose with covariance subscriber
+    rclcpp::Subscription<geometry_msgs::msg::PoseStamped>::SharedPtr sub_pose_;                          //!< @brief measurement pose subscriber
+    rclcpp::Subscription<geometry_msgs::msg::TwistWithCovarianceStamped>::SharedPtr sub_twist_with_cov_; //!< @brief measurement twist with covariance subscriber
+    rclcpp::Subscription<geometry_msgs::msg::TwistStamped>::SharedPtr sub_twist_;                        //!< @brief measurement twist subscriber
+
+    rclcpp::TimerBase::SharedPtr timer_control_;             //!< @brief time for ekf calculation callback
+    // tf2_ros::TransformBroadcaster tf_br_;  //!< @brief tf broadcaster
+
+    TimeDelayKalmanFilter ekf_;  //!< @brief  extended kalman filter instance.
+
+    /* parameters */
+    double proc_stddev_yaw_c_ = 0.005, proc_stddev_yaw_bias_c_ = 0.001, proc_stddev_vx_c_ = 2.0, proc_stddev_wz_c_ = 0.2;
```

Comment:
Most of these autoware packages dont have a config parameters file, we are directly defining the values in the launch file in carma.
I left the original implementation in so that there's no confusion about where the param is being loaded from. 
I'm ok with moving though, if you think it would be helpful

Reply:
Sorry - I missed that these actually were being read in by the node with declare_parameter and get_parameter, and it looks like these values can be adjusted through localization.launch.py in carma-platform without needing to rebuild. I'll mark this as resolved!